File: Cpp/1221_Split_a_String_in_Balanced_Strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int result = 0;
        int f = 0; //s[0] == 'R'? 1 : -1;
        for(int i = 0; i < s.size(); i++){
            f += s[i] == 'R'? 1:-1;
            if(f == 0){
                result++;
            }
        }
        return result;
    }
};

File: Java/1221_Split_a_String_in_Balanced_Strings.java
class Solution {
    public int balancedStringSplit(String s) {
        int result = 0;
        int r = 0;
        int l = 0;
        for(int i = 0; i < s.length(); i++){
            
            switch(s.charAt(i)){
                case 'R':
                    r++;
                    break;
                case 'L':
                    l++;
                    break;
            }
            
            result += (r == l)? 1: 0;
            /*
            if(r == l){
                result++;
            }
            */
        }
        return result;
    }
}
